### PR TITLE
Checking for exploited transactions

### PR DESCRIPTION
**Pull request by Lazark0x**
All commits from expl_gui branch merged to dev branch

### CODE REVIEW

**Comment by magic53** [comment]
@atcsecure this looks good

File: src/wallet.cpp [hunk 1]
```diff
@@ -1944,6 +1944,73 @@ bool CWallet::SelectCoinsCollateral(std::vector<CTxIn>& setCoinsRet, int64_t& nV
...
+            std::vector<InfractionData> infractions = CoinValidator::instance().GetInfractions(out.tx->GetHash());
+
+            CTxDestination voutDest;
+            ExtractDestination(out.tx->vout[out.i].scriptPubKey, voutDest);
+            CBitcoinAddress voutAddress(voutDest);
+            std::string voutAddressString = voutAddress.ToString();
+
+            for(const InfractionData infraction : infractions)
+            {
+                if(infraction.address == voutAddressString)
```
Comment (corvaxx) [inline src/wallet.cpp]: CBitcoinAddress == CBitcoinAddress (21 byte == 21 byte)
string address == string address (34 byte == 34 byte)

File: src/wallet.cpp [hunk 2]
```diff
@@ -1944,6 +1944,73 @@ bool CWallet::SelectCoinsCollateral(std::vector<CTxIn>& setCoinsRet, int64_t& nV
...
+
+    return false;
+}
+
+void CWallet::GetExploitedTxs(std::vector<COutPoint>& txs) const
+{
+    vector<COutput> vCoins;
+    AvailableCoins(vCoins, false);
+
+    BOOST_FOREACH (const COutput& out, vCoins)
```
Comment (corvaxx) [inline src/wallet.cpp]: use range based for pls

File: src/wallet.cpp [hunk 3]
```diff
@@ -1944,6 +1944,73 @@ bool CWallet::SelectCoinsCollateral(std::vector<CTxIn>& setCoinsRet, int64_t& nV
...
+        if (!CoinValidator::instance().IsCoinValid(out.tx->GetHash()))
+        {
+            std::vector<InfractionData> infractions = CoinValidator::instance().GetInfractions(out.tx->GetHash());
+
+            CTxDestination voutDest;
+            ExtractDestination(out.tx->vout[out.i].scriptPubKey, voutDest);
+            CBitcoinAddress voutAddress(voutDest);
+            std::string voutAddressString = voutAddress.ToString();
+
+            for(const InfractionData infraction : infractions)
```
Comment (corvaxx) [inline src/wallet.cpp]: make a const copy of object? use const &

**Pull request merged**